[ARROW][GH-34147] Adding testfile for dictionary and crc (#37)
apache/parquet-testing: data/README.md

| bloom_filter.bin                               | deprecated bloom filter binary with binary header and murmur3 hashing |
| bloom_filter.xxhash.bin                        | bloom filter binary with thrift header and xxhash hashing    |
| nan_in_stats.parquet                           | statistics contains NaN in max, from PyArrow 0.8.0. See note below on "NaN in stats".  |

> | rle-dict-snappy-checksum.parquet                 | compressed INT32 and STRING columns in v2 dictionary pages with a matching CRC |

Reviewer: Does it matter whether the file format is v1 or v2? It seems to me that they are the same dictionary pages except the encoding name. I'd suggest to add less files to keep this repo small.

Author: Okay, previously, I though we can test format-V1 and format-V2 files. But exactly, v2 is enough for us.

| bloom_filter.bin                               | deprecated bloom filter binary with binary header and murmur3 hashing |
| bloom_filter.xxhash.bin                        | bloom filter binary with thrift header and xxhash hashing    |
| nan_in_stats.parquet                           | statistics contains NaN in max, from PyArrow 0.8.0. See note below on "NaN in stats".  |

> | rle-dict-snappy-checksum.parquet                 | compressed INT32 and STRING columns in v2 dictionary pages with a matching CRC |
> | plain-dict-uncompressed-checksum.parquet         | uncompressed INT32 and STRING columns in v1 dictionary pages with a matching CRC |
> | rle-dict-uncompressed-corrupt-checksum.parquet   | uncompressed INT32 and STRING columns in v2 dictionary pages with a mismatching CRC |

Reviewer: ```suggestion
| rle-dict-snappy-checksum.parquet                 | compressed and dictionary-encoded INT32 and STRING columns in format v2 with a matching CRC |
| plain-dict-uncompressed-checksum.parquet         | uncompressed and dictionary-encoded INT32 and STRING columns in format v1 with a matching CRC |
| rle-dict-uncompressed-corrupt-checksum.parquet   | uncompressed and dictionary-encoded INT32 and STRING columns in format v2 with a mismatching CRC |
```

  [ Column "b" [ Page 0 [correct crc] | Uncompressed Contents ][ Page 1 [bad crc] | Uncompressed Contents ]]
  ```

The schema for the `*-dict-*-checksum.parquet` test files is:
* `data/rle-dict-snappy-checksum.parquet`:
  ```

>   [ Column "long_field" [ Dict Index Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
>   [ Column "binary_field" [Dict Index Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]

Reviewer: ```suggestion
  [ Column "long_field" [ Dict Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
  [ Column "binary_field" [ Dict Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
```

  ```

* `data/plain-dict-uncompressed-checksum.parquet`:
  ```

>   [ Column "long_field" [ Dict Index Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Unompressed PLAIN_DICTIONARY Contents ]]
>   [ Column "binary_field" [Dict Index Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Uncompressed PLAIN_DICTIONARY Contents ]]

Reviewer: ```suggestion
  [ Column "long_field" [ Dict Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Uncompressed PLAIN_DICTIONARY Contents ]]
  [ Column "binary_field" [ Dict Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Uncompressed PLAIN_DICTIONARY Contents ]]
```

  ```

* `data/rle-dict-uncompressed-corrupt-checksum.parquet`:
  ```

>   [ Column "long_field" [ Dict Index Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
>   [ Column "binary_field" [Dict Index Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]

Reviewer: ```suggestion
  [ Column "long_field" [ Dict Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
  [ Column "binary_field" [ Dict Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
```
